src/Components/Web.JS/src/Boot.Web.ts

```diff
@@ -1,25 +1,48 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
 // Currently this only deals with inserting streaming content into the DOM.
 // Later this will be expanded to include:
 //  - Progressive enhancement of navigation and form posting
 //  - Preserving existing DOM elements in all the above
 //  - The capabilities of Boot.Server.ts and Boot.WebAssembly.ts to handle insertion
 //    of interactive components
 
+import { startCircuit } from './Boot.Server.Common';
+import { startWebAssembly } from './Boot.WebAssembly.Common';
 import { shouldAutoStart } from './BootCommon';
+import { Blazor } from './GlobalExports';
+import { WebStartOptions } from './Platform/WebStartOptions';
 import { attachStreamingRenderingListener } from './Rendering/StreamingRendering';
+import { WebAssemblyComponentDescriptor } from './Services/ComponentDescriptorDiscovery';
+import { ServerComponentDescriptor, discoverComponents } from './Services/ComponentDescriptorDiscovery';
 
 let started = false;
 
-async function boot(): Promise<void> {
+async function boot(options?: Partial<WebStartOptions>): Promise<void> {
   if (started) {
     throw new Error('Blazor has already started.');
   }
   started = true;
+  await activateInteractiveComponents(options);
 
   attachStreamingRenderingListener();
 }
 
-window['Blazor'] = { start: boot }; // Temporary API stub until we include interactive features
+async function activateInteractiveComponents(options?: Partial<WebStartOptions>) {
+  const serverComponents = discoverComponents(document, 'server') as ServerComponentDescriptor[];
+  const webAssemblyComponents = discoverComponents(document, 'webassembly') as WebAssemblyComponentDescriptor[];
```

Comment:
Currently, "auto" mode will cause an exception to be thrown because we don't currently handle the case where a component has multiple descriptors outputted.